lib/commands/device/common.js

```diff
@@ -136,6 +141,10 @@ export async function getDeviceInfoFromCaps() {
  * @returns {Promise<import('appium-adb').ADB>}
  */
 export async function createADB() {
+  if (this.opts.adbListenAllNetwork) {
```

Comment:
should we instead move this check into createBaseADB to avoid duplication?

Reply:
Let me tweak further. `this` didn't work with the current implementation in the module so moved to this way